docs/decisions/0006-pluggable-lti-configuration.rst

```diff
@@ -0,0 +1,116 @@
+Pluggable and re-usable LTI configuration
+-----------------------------------------
+
+
+Status
+======
+
+Proposal
+
+Context
+=======
+
+LTI Tools provide extra funcionality and allows course creators to embed external interactive content
+in the LMS through a standardized setup.
+
+Currently in the Open edX platform:
+
+#. Each LTI tool needs to be configured individually, sometimes requiring multiple valiables to be set to get
+a working integration. This makes the course creation process cumbersome on courses that heavily rely on LTI 
+components. 
+#. There's no support for instance-wide LTI tool configuration or any sort of pluggable LTI configurations.
+
+
+This proposal aims to solve these issues by introducing a mechanism to allow LTI configuration pluggability 
+and changes to the XBlock to allow re-using configurations to reduce instructor friction.
+
+
+Decision
+========
+
+The decision is to enable LTI reusability and pluggability through a combination of a filter hook (as defined in `OEP-50`_) 
+and a standardized API interface. 
+Through the API interface, plugins should be able to set up externally managed LTI configurations store their data, while
+the LTI consumer handles displaying the content.
+
+To enable this, the LTI Configuration model needs to be modified to:
+
+#. Allow externally configured tools (a new {{CONFIG_EXTERNAL}} choice for the configuration storage).
+#. Two new fields to store an unique external tool id and the plugin name/location.
+#. Changes to the {{get_lti_consumer}} methods to allow instancing a {{LtiConsumer}} from externally managed configurations.
+
+
+The filter part will allow the XBlock LTI Consumer to query re-usable and/or plugabble LTI configurations and display them to course 
+creators as configuration options in Studio.
+
+The filter response should contain all the necessary information for the XBlock to locate and use the externally managed LTI configuration.
+The reponse **should** be structured this way:
+
+.. code-block:: json
+    {
+        "filter_name": {
+            "api_path": "plugin_1_module_name.api.get_lti_config",
+            "available_configurations": [
+                {
+                    "lti_version": "LTI_1P1",
+                    "display_name": "Reusable LTI tool 1",
+                    "lti_config_id": 2,
+                }
+                {
+                    "lti_version": "LTI_1P3",
+                    "display_name": "Reusable LTI tool 2 - LTI 1.3",
+                    "lti_config_id": 3,     
+                }
+            ]
+        }
+    }
+
+
+Each LTI configuration provider MUST:
+
+* Provide a `get_lti_config` method through the `api_path` parameter conforming with the API parameters described in the `Plugin API interface`.
```

Comment:
@xitij2000 What I had in mind here was to store a method path in the `LtiConfiguration` models and use `importlib` to call it with well-specified interface that'll describe below.